### DISCUSSION TITLE
Welcome to Mago Discussions!

### DISCUSSION THREAD

**Discussion by azjezz** [carthage-software/mago#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by haringsrob** [comment]
Hey 👋 
Just curious. I tried it briefly and seems to work great so far.

What are the long term plans? Is there going to be a language server for php build on top of this (given parsing etc is done in rust, we could finally have a performant lsp with project wide diagnostics a'la phpstan).

Might be able to help (a bit).

Cheers.

**Comment by azjezz** [comment]
Hi! 

An LSP is planned, but it is not our top priority right now, the current roadmap is to get the linter and formatter to stable state, and implement a type checker, then we can have an LSP.

There's already a PoC LSP implementation here: https://github.com/carthage-software/mago/pull/74
